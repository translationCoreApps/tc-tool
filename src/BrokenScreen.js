import React from 'react';
import PropTypes from 'prop-types';

const styles = {
  root: {
    width: '100%',
    margin: '40px'
  },
  title: {
    textAlign: 'center',
    width: '100%'
  },
  code: {
    color: '#cc3939',
    backgroundColor: '#ffe6e6',
    lineHeight: '150%',
    whiteSpace: 'pre-wrap'
  }
};

/**
 * Renders an error screen
 * @param {string} title - the title of the broken screen
 * @param {*} error - the error object
 * @param {*} info - the info object
 * @return {*}
 * @constructor
 */
const BrokenScreen = ({title, error, info}) => {
  return (
    <div style={styles.root}>
      <h1 style={styles.title}>{title}</h1>
      <details style={styles.code}>
        {error && error.toString()}
        <br/>
        {info.componentStack}
      </details>
    </div>
  );
};

BrokenScreen.propTypes = {
  title: PropTypes.string.isRequired,
  error: PropTypes.any.isRequired,
  info: PropTypes.any.isRequired
};

export default BrokenScreen;
