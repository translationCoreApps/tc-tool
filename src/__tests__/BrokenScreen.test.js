import React from 'react';
import renderer from 'react-test-renderer';
import BrokenScreen from '../BrokenScreen';

test('snapshot', () => {
    const info = { componentStack: `in WordBankItem
    in DragSource(WordBankItem)
    in div
    in WordBankArea` };
    const error = "Error: Some error message";
    const wrapper = renderer.create(
        <BrokenScreen title="title" info={info} error={error}/>
    );
    expect(wrapper).toMatchSnapshot();
});