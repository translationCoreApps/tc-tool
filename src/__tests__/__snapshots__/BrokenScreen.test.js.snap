// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  style={
    Object {
      "margin": "40px",
      "width": "100%",
    }
  }
>
  <h1
    style={
      Object {
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    title
  </h1>
  <details
    style={
      Object {
        "backgroundColor": "#ffe6e6",
        "color": "#cc3939",
        "lineHeight": "150%",
        "whiteSpace": "pre-wrap",
      }
    }
  >
    Error: Some error message
    <br />
    in WordBankItem
    in DragSource(WordBankItem)
    in div
    in WordBankArea
  </details>
</div>
`;
